drivers/gpio/gpio_sifive.c

```diff
@@ -116,93 +116,26 @@ static int gpio_sifive_config(struct device *dev,
 		return -EINVAL;
 	}
 
-	/* Configure gpio direction */
-	if (flags & GPIO_DIR_OUT) {
-		gpio->in_en &= ~BIT(pin);
-		gpio->out_en |= BIT(pin);
-
-		/*
-		 * Account for polarity only for GPIO_DIR_OUT.
-		 * invert register handles only output gpios
-		 */
-		if (flags & GPIO_POL_INV) {
-			gpio->invert |= BIT(pin);
-		} else {
-			gpio->invert &= ~BIT(pin);
-		}
-	} else {
-		gpio->out_en &= ~BIT(pin);
-		gpio->in_en |= BIT(pin);
-
-		/* Polarity inversion is not supported for input gpio */
-		if (flags & GPIO_POL_INV) {
-			return -EINVAL;
-		}
-
-		/*
-		 * Pull-up can be configured only for input gpios.
-		 * Only Pull-up can be enabled or disabled.
-		 */
-		if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_DOWN) {
-			return -EINVAL;
-		}
-
-		if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_UP) {
-			gpio->pue |= BIT(pin);
-		} else {
-			gpio->pue &= ~BIT(pin);
-		}
+	/* We cannot support open-source open-drain configuration */
+	if ((flags & GPIO_SINGLE_ENDED) != 0) {
+		return -ENOTSUP;
 	}
 
-	/*
-	 * Configure interrupt if GPIO_INT is set.
-	 * Here, we just configure the gpio interrupt behavior,
-	 * we do not enable/disable interrupt for a particular
-	 * gpio.
-	 * Interrupt for a gpio is:
-	 * 1) enabled only via a call to gpio_sifive_enable_callback.
-	 * 2) disabled only via a call to gpio_sifive_disabled_callback.
-	 */
-	if (!(flags & GPIO_INT)) {
-		return 0;
+	/* We only support pull-ups, not pull-downs */
+	if ((flags & GPIO_PULL_DOWN) != 0) {
+		return -ENOTSUP;
 	}
 
-	/*
-	 * Interrupt cannot be set for GPIO_DIR_OUT
-	 */
-	if (flags & GPIO_DIR_OUT) {
-		return -EINVAL;
-	}
+	/* Set the initial output value before enabling output to avoid
+	 * glitches */
+	WRITE_BIT(gpio->out_val, pin, flags & GPIO_OUTPUT_INIT_HIGH);
```

Comment:
This part we need to write differently since the API supports a case where the output value is not initialized before enabling it. One way to do it would be
```
if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0) {
	gpio->out_val |= BIT(pin);
}
if ((flags & GPIO_OUTPUT_INIT_LOW) != 0) {
	gpio->out_val &= ~BIT(pin);
}
```
I would also move this code section below line 134. Then pull up / pull down configuration and output / input would be next to each other.